core/geometry/src/test/curve/CurveCurveCloseApproachXY.test.ts

```diff
@@ -885,6 +888,115 @@ describe("CurveCurveCloseApproachXY", () => {
     GeometryCoreTestIO.saveGeometry(allGeometry, "CurveCurveCloseApproachXY", "SingleArcArc4");
     expect(ck.getNumErrors()).equals(0);
   });
+  function test2Ellipses(
+    ck: Checker,
+    allGeometry: GeometryQuery[],
+    geometryA: Arc3d,
+    geometryB: Arc3d,
+    numExpectedIntersections: number,
+    numExpectedPerpCloseApproach: number,
+  ) {
+    const existsInIntersectionSetWithTolerance = (set: Set<Point3d>, point: Point3d, tolerance: number = 1e-6): boolean => {
+      for (const pt of set)
+        if (pt.isAlmostEqual(point, tolerance))
+          return true;
+      return false;
+    };
+
+    const existsInApproachSetWithTolerance = (
+      set: Set<[Point3d, Point3d]>, point0: Point3d, point1: Point3d, tolerance: number = 1e-6,
```

Comment:
For future reference: you can avoid this lambda and simplify your code by instead using an `OrderedSet` constructed with a toleranced comparison function.  `OrderedSet` will do all this work for you in its `add` method: an element won't get added to the set unless it's different according to your compare function.

Reply:
good to know